session/pessimistic_test.go

```diff
@@ -2025,26 +2025,37 @@ func (s *testPessimisticSuite) TestAsyncCommitWithSchemaChange(c *C) {
 	tk3 := testkit.NewTestKitWithInit(c, s.store)
 
 	// The txn tk writes something but with failpoint the primary key is not committed.
-	tk.MustExec("begin optimistic")
+	tk.MustExec("begin pessimistic")
 	tk.MustExec("insert into tk values(2, 2, 2)")
-	// Add index for c2 before commit
-	tk2.MustExec("alter table tk add index k2(c2)")
+	tk.MustExec("update tk set c2 = 10 where c1 = 1")
+	ch := make(chan struct{})
+	go func() {
+		// Add index for c2 before commit
+		tk2.MustExec("alter table tk add index k2(c2)")
+		ch <- struct{}{}
+	}()
```

Comment:
May sleep for a short duration waiting for the statement executed.

Reply:
Fixed. PTAL again.